Description: Backport PR #4758 - fixes for Python 3.4.
 Not complete, plenty more failures
Author: Thomas Kluyver <takowl@gmail.com>
Origin: 341c046c7ce22c633daf3cb12f3f439c6f376056

--- a/IPython/utils/tests/test_text.py
+++ b/IPython/utils/tests/test_text.py
@@ -15,6 +15,7 @@
 import os
 import math
 import random
+import sys
 
 import nose.tools as nt
 
@@ -107,7 +108,12 @@ def eval_formatter_no_slicing_check(f):
     s = f.format('{stuff[slice(1,4)]}', **ns)
     nt.assert_equal(s, 'ell')
     
-    nt.assert_raises(SyntaxError, f.format, "{a[:]}")
+    if sys.version_info >= (3, 4):
+        # String formatting has changed in Python 3.4, so this now works.
+        s = f.format("{a[:]}", a=[1, 2])
+        nt.assert_equal(s, "[1, 2]")
+    else:
+        nt.assert_raises(SyntaxError, f.format, "{a[:]}")
 
 def test_eval_formatter():
     f = text.EvalFormatter()
